streaming/java/streaming-state/src/main/java/org/ray/streaming/state/keystate/state/impl/AbstractState.java

```diff
@@ -0,0 +1,51 @@
+package org.ray.streaming.state.keystate.state.impl;
+
+import com.google.common.base.Preconditions;
+import org.ray.streaming.state.backend.TransactionKeyStateBackend;
+import org.ray.streaming.state.keystate.desc.AbstractStateDescriptor;
+
+/**
+ * Basic State.
+ */
+public abstract class AbstractState<T> {
```

Comment:
Now we have `State` interface and `AbstractState` which doesn't have inheritance or directly combination. Why?

Reply:
Yes, that's a little confusing, I'll reconsider the relationship between them